### DISCUSSION TITLE
Drop first argument in `withParent` functions

### DISCUSSION THREAD

**Discussion by St4B** [LemonAppDev/konsist#871] [Request a Feature]
### Tell us more.

All the `withParent` functions have a `KClass` argument and then we can pass more parents with the help of `vararg`. I suppose that this happens to enforce passing at least one parent. Maybe it would be better to drop this argument and add an internal validation or in case of empty parents return all the classes. Basically, I would like to be able to write something like that:

```kotlin
val declarations = setOf(First::class, Second::class, Third::class)

Konsist.scopeFromProject()
        .interfaces()
        .withParentOf(*declarations.toTypedArray())
```            

Something like that will work but seems extra work without any actual value:

```kotlin
val declarations = setOf(First::class, Second::class, Third::class)

Konsist.scopeFromProject()
        .interfaces()
        .withParentOf(declarations.first(), *declarations.drop(1).toTypedArray())
```            

I think the first API seems cleaner. WDYT?

**Comment by igorwojda** [comment]
Great idea. We will add it to Konsist at some point.
I have created a ticket https://lemonappdev.atlassian.net/browse/KON-598

**Comment by igorwojda** [comment]
API has been updated in [Konsist 0.15.0](https://github.com/LemonAppDev/konsist/releases/tag/v0.15.0). Now collections can be passed as arguments.
